docs/jupyter-widgets.qmd

```diff
@@ -0,0 +1,263 @@
+map.center = (51, 0)
+```
+
+With shinywidgets, we can do the same property updates _reactively_ in Shiny.
+The main thing to know is to access the widget instance through the `widget` property of the `@render_widget`-decorated function.
+For example, the following code creates a map, then updates the map's center whenever the dropdown changes.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+#| viewerHeight: 400
+from shiny import ui, reactive
+from shiny.express import input
+from shinywidgets import render_widget
+import ipyleaflet as ipyl
+
+city_centers = {
+    "London": (51.5074, 0.1278),
+    "Paris": (48.8566, 2.3522),
+    "New York": (40.7128, -74.0060)
+}
+
+ui.input_select("center", "Center", choices=list(city_centers.keys()))
+
+@render_widget
+def map():
+    return ipyl.Map(zoom=4)
+
+@reactive.effect
+def _():
+    map.widget.center = city_centers[input.center()]
+## file: requirements.txt
+ipyleaflet
+```
+
+::: callout-note
+## Re-render vs update in-place
+
+If the app above had used `@render_widget` instead of `@reactive.effect` to perform the update, then the map gets re-rendered from stratch every time `input.center` changes, which causes the map to flicker (instead of a smooth transition to the new location).
+:::
+
+
+
+## Access user interaction {#access-user-interaction}
+
+There are two main ways to access user interaction with ipywidgets in Shiny:
+
+1. [Reactive properties](#reactive-read)
+2. [Capturing widget events](#capturing-widget-events)
+
+Assuming the user interaction you're interested in is available through a widget property, then the first approach is usually the best/easiest choice.
+
+
+### Reactive properties {#reactive-read}
+
+If you've used ipywidgets before, you may know that widgets have [properties](https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20Basics.html#widget-properties) (aka "traitlets") that can be accessed and observed.
+For example, in a notebook, you may have written a code cell like this to first display a map:
+
+```python
+import ipyleaflet as ipyl
+map = ipyl.Map()
+```
+
+Then, in a later cell, you may have read the map's `center` property to get the current map's location:
+
+```python
+map.center
+```
+
+With shinywidgets, we can do the same property reads _reactively_ in Shiny.
+The main thing to know is to access the widget instance through the `widget` property of the `@render_widget`-decorated function.
+For example, the following example creates a `map`, then displays the current `center` location via `@render.text`:
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+#| viewerHeight: 450
+import ipyleaflet as ipyl
+from shinywidgets import reactive_read, render_widget
+from shiny import render
+import shiny.express
+
+@render_widget
+def map():
+    return ipyl.Map(zoom=2)
+
+@render.text
+def center():
+    cntr = reactive_read(map.widget, 'center')
```

Comment:
TODO: Finish https://github.com/posit-dev/py-shinywidgets/pull/119 and update/proof-read writing